### PR TITLE
Add git_user_name and git_user_email option to match

### PR DESCRIPTION
**Pull request by kouki-dan**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

Add new option to match `git_user_name` and `git_user_email` to set git config to match's git repository.

### Motivation and Context

If using useConfigOnly option (and global name or email settings is not set) in gitconfig, match fails to commit to git repository
Add new two option for name and email and fix it.

The reason which the git option needs is followed by [this page](https://collectiveidea.com/blog/archives/2016/04/04/multiple-personalities-in-git)

### CODE REVIEW

**Comment by kouki-dan** [comment]
I signed it!

**Comment by kouki-dan** [comment]
I signed it!

File: match/lib/match/options.rb [hunk 1]
```diff
@@ -64,6 +64,16 @@ def self.available_options
                                      verify_block: proc do |value|
                                        ENV["FASTLANE_TEAM_ID"] = value.to_s
                                      end),
+        FastlaneCore::ConfigItem.new(key: :git_user_name,
+                                     env_name: "MATCH_GIT_USER_NAME",
+                                     description: "git user name to commit",
```
Comment (KrauseFx) [inline match/lib/match/options.rb]: Maybe that's just a small detail, but by `user name` many people understand the nickname (in my case `KrauseFx`), however in git it's recommended to use the full name as far as I know (in my case `Felix Krause`). Maybe we should rename the `key`, `env_name` and `description` to clarify this?

**Review by KrauseFx — CHANGES_REQUESTED** [review]

**Comment by KrauseFx** [comment]
Hey @kouki-dan, thanks for taking the time to submit this PR, a great change, and I can totally see the use case for this 👍

File: match/lib/match/options.rb (hunk 1 above)
Comment (kouki-dan) [inline match/lib/match/options.rb]: Thank you for your review. I understand.
I will replace `user name` to `full name` and push this.

File: match/lib/match/git_helper.rb [hunk 2]
```diff
@@ -136,5 +138,22 @@ def self.copy_readme(directory)
...
       File.write(File.join(directory, "README.md"), template)
     end
+
+    def self.add_user_config(user_name, user_email)
+      # Add git config if needed
+      commands = []
+      commands << "git config user.name \"#{user_name}\"" unless user_name.nil?
+      commands << "git config user.email \"#{user_email}\"" unless user_email.nil?
+
+      UI.message "Add git user config to local git repo..." unless commands.empty?
```
Comment (KrauseFx) [inline match/lib/match/git_helper.rb]: Maybe you can prepend this with a line that's just
```ruby
return if commands.empty?
```

Comment (kouki-dan) [inline match/lib/match/git_helper.rb]: Ah, okay. That's better I think.

**Review by KrauseFx — APPROVED** [review]
This looks great 🎷

**Comment by KrauseFx** [comment]
Thanks for your contribution @kouki-dan 👍

**Comment by hjanuschka** [comment]
i think that fixes: https://github.com/fastlane/fastlane/issues/8450

**Comment by kouki-dan** [comment]
Yes, I got a same error as #8450.

Thank you for your reviewing and merging. I support Fastlane project :+1:

**Comment by kylejm** [comment]
This is an awesome improvement! This problem got me for ages. I had to keep changing my gitconfig back and forth which was error prone. Thank you!

However, I'm not sure why, but this isn't working for me. I'm running fastlane `2.28.7`.

Tried:

```shell 
$ fastlane match nuke distribution --skip_confirmation -a my.app.id --git_full_name 'Fastlane Match' --git_user_email 'match@fastlane.tools'
```

and 

```ruby
#Fastfile
match(app_identifier: identifier,
          readonly: readonly,
          type: type,
          shallow_clone: 1,
          git_full_name: 'Fastlane Match',
          git_user_email: 'match@fastlane.tools')
```

Any idea as to what I'm doing wrong?

**Comment by KrauseFx** [comment]
@kylejm that's how you should use it. Can you submit a new issue with the output? Is it not working?

**Comment by kylejm** [comment]
Hey @KrauseFx, sorry for the slow response. Issue create ☝️.

**Pull request merged**